### ISSUE TITLE
Class not found exception

### ISSUE THREAD

**Issue by zhanyigo** [cbornet/generator-jhipster-grpc#20]
## Steps:

1. yo jhipster
2. import jdl
3. generate grpc
4. run with mvn

## First problem:

```bash
[ERROR] /Users/zhanyi/m2/target/generated-sources/protobuf/java/com/xxx/dcm/grpc/PageRequestAndFiltersOrBuilder.java:[26,36] 无法访问com.xxx.dcm.grpc.QueryFilter
  错误的源文件: /Users/zhanyi/m2/target/generated-sources/protobuf/java/com/xxx/dcm/grpc/QueryFilter.java
    文件不包含类com.xxx.dcm.grpc.QueryFilter
    请删除该文件或确保该文件位于正确的源路径子目录中。
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 7.249 s
[INFO] Finished at: 2018-05-07T09:06:37+08:00
[INFO] Final Memory: 46M/359M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project m-2: Compilation failure
[ERROR] /Users/zhanyi/m2/target/generated-sources/protobuf/java/com/xxx/dcm/grpc/PageRequestAndFiltersOrBuilder.java:[26,36] 无法访问com.xxx.dcm.grpc.QueryFilter
[ERROR] 错误的源文件: /Users/zhanyi/m2/target/generated-sources/protobuf/java/com/xxx/dcm/grpc/QueryFilter.java
[ERROR] 文件不包含类com.xxx.dcm.grpc.QueryFilter
[ERROR] 请删除该文件或确保该文件位于正确的源路径子目录中。
[ERROR] -> [Help 1]
```

### I Fixed it

The QueryFilter.java contain a class `Queryfilter` not `QueryFilter`.
I add the code to the `src/main/proto/util/queryfilter.proto`.  The first problem was gone. And I got the second problem.
```
option java_outer_classname = "QueryFilterGrpc";
```

### The errors:

```java
[ERROR] /Users/zhanyi/m2/src/main/java/com/xxx/dcm/grpc/ProtobufMappers.java:[122,19] 找不到符号
  符号:   类 AuditEvent
  位置: 类 com.xxx.dcm.grpc.ProtobufMappers
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 8.263 s
[INFO] Finished at: 2018-05-07T09:15:36+08:00
[INFO] Final Memory: 49M/448M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project m-2: Compilation failure
[ERROR] /Users/zhanyi/m2/src/main/java/com/xxx/dcm/grpc/ProtobufMappers.java:[122,19] 找不到符号
[ERROR] 符号:   类 AuditEvent
[ERROR] 位置: 类 com.xxx.dcm.grpc.ProtobufMappers
[ERROR] -> [Help 1]
[ERROR]
```

## yo jhipster:info

```bash

Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
m-2@0.0.0 /Users/zhanyi/m2
└── generator-jhipster@4.14.1

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.xxx.dcm",
      "nativeLanguage": "zh-cn"
    },
    "jhipsterVersion": "4.14.1",
    "baseName": "m2",
    "packageName": "com.xxx.dcm",
    "packageFolder": "com/xxx/dcm",
    "serverPort": "8080",
    "authenticationType": "oauth2",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "nativeLanguage": "zh-cn",
    "languages": [
      "zh-cn"
    ],
    "clientFramework": "angularX",
    "useSass": false,
    "clientPackageManager": "yarn"
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
/**
 * 成交信息
 */
entity Deal (deal) {
  /**
   * 成交编号
   */
  dealNo String,
  tradeMethod Integer,
  price BigDecimal,
  amount BigDecimal,
  bidParty Integer,
  bidOrder Long,
  offerParty Integer,
  offerOrder Long,
  timestamp Instant,
  status DealStatus
}
entity Market (market) {
 
}
/**
 * The Contract entity.
 */
entity Crontact (crontact) {
  
}

entity ContractParameter (contract_parameter) {
  
}

entity Currency (currency) {
 
}
/**
 * The TradeAccount entity.
 */
entity TradeAccount (trade_account) {
  /**
   * User Id
   */
  userId String,
  status Status
}

enum DealStatus {
  DONE,
  PENDING,
  BINDED
}

enum Status {
  PENDDING,
  ENABLED,
  DISABLED,
  EXPIRED
}

enum ContractType {
  
}

enum CurrencyType {
 
}

enum BlockChainPlatform {
 
}

relationship OneToOne {
  ContractParameter{contract} to Crontact
}
relationship OneToMany {
  TradeAccount{creditLimts} to CreditLimits{tradeAccount}
}
relationship ManyToOne {
  CreditLimits{currency} to Currency
}

dto Deal, Market, Crontact, ContractParameter, Currency, TradeAccount, CreditLimits with mapstruct
paginate Deal with infinite-scroll
service Deal, Market, Crontact, ContractParameter, Currency, TradeAccount, CreditLimits with serviceImpl
angularSuffix Deal, Market, Crontact, ContractParameter, Currency, TradeAccount, CreditLimits with agl
paginate Crontact, ContractParameter, Currency with pagination

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_152"
Java(TM) SE Runtime Environment (build 1.8.0_152-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.152-b16, mixed mode)

git version 2.15.1 (Apple Git-101)

node: v8.11.1

npm: 6.0.0

bower: 1.8.4

gulp:
[09:03:17] CLI version 2.0.1

yeoman: 2.0.2

yarn: 1.5.1

Docker version 17.12.0-ce, build c97c6d6

docker-compose version 1.18.0, build 8dd22a9

```

**Comment by cbornet** [comment]
Yes this is a weird issue that only occurs on MAC OS. It is fixed in the `next_jhipster` branch which targets JHipster 5. Do you need it to be fixed for JHipster 4 ?

**Comment by zhanyigo** [comment]
Can you fix it?  Jhipster 5 need some time .

**Comment by cbornet** [comment]
This should be fixed in version 0.15.2. Can you try ?
